source/common/upstream/load_balancer_impl.cc

```diff
@@ -1299,6 +1299,33 @@ HostConstSharedPtr LeastRequestLoadBalancer::unweightedHostPick(const HostVector
                                                                 const HostsSource&) {
   HostSharedPtr candidate_host = nullptr;
 
+  // Use a full scan if:
+  // - it's explicitly configured, or
+  // - the number of choices is equal to or larger than the number of hosts.
+  const auto has_full_scan_configured = selection_method_ ==
+                                        envoy::extensions::load_balancing_policies::least_request::v3::LeastRequest_SelectionMethod_FULL_SCAN;
+  if (has_full_scan_configured || (hosts_to_use.size() <= choice_count_)) {
```

Comment:
Can you pull this block out into its own function?  It would probably look nice if you had a switch statement that calls the appropriate selection function.